3/3.js

```diff
@@ -0,0 +1,222 @@
+    {
+        id: 2,
+        name: "Тройной чизбургер",
+        price: 2.3,
+        currency: "usd",
+        ingredients: ["flour", "beef", "cheese", "sauce", "cucumber"],
+        type: "burger",
+        available: true
+    },
+    {
+        id: 3,
+        name: "Чизбургер",
+        price: 1,
+        currency: "euro",
+        ingredients: ["flour", "beef", "cheese", "sauce", "cucumber"],
+        type: "burger",
+        available: true
+    },
+    {
+        id: 4,
+        name: "Картофель фри",
+        price: 2,
+        currency: "usd",
+        ingredients: ["potato"],
+        type: "snack",
+        available: true
+    },
+    {
+        id: 5,
+        name: "Картофель по-деревенски",
+        price: 2,
+        currency: "usd",
+        ingredients: ["potato"],
+        type: "snack",
+        available: false
+    },
+    {
+        id: 6,
+        name: "Чикен МакНаггетс™",
+        price: 5,
+        currency: "euro",
+        ingredients: ["chicken"],
+        type: "chicken",
+        available: true
+    },
+    {
+        id: 7,
+        name: "Стрипсы",
+        price: 2.6,
+        currency: "euro",
+        ingredients: ["chicken"],
+        type: "chicken",
+        available: false
+    },
+    {
+        id: 8,
+        name: "МакЧикен",
+        price: 4.3,
+        currency: "euro",
+        ingredients: ["chicken", "flour", "cheese", "sauce"],
+        type: "burger",
+        available: false
+    },
+
+
+
+]
+
+
+//1. Собрать в массив ингредиентов (без повторения).
+
+const allIngredients = data.reduce((acc, item) => {
+    const ingredients = item.ingredients
+    acc.push(ingredients)
+    return acc
+}, [])
+const allAccIngredients = allIngredients.join().split(',')
+const uniqIngredients = allAccIngredients.filter((item, index) => allAccIngredients.indexOf(item) === index);
+console.log(uniqIngredients);
+
+
+//2. Создать функцию, которая принимает массив продуктов и id, 
+//и возвращает продукт с таким же id.
+
+const findProduct = function(data, id) {;
+
+
+        const product = data.find(product => product.id === id)
+            //let product = cities.find((city) => {
+            //    return city.name === searchTerm).id}
+        return product
+    }
+    //console.log(findProduct(data, 8));
+
+
+//3. Создать массив с отсортированными продуктами по цене.
+
+
+function sortByPrice(data) {
+    return data.sort((a, b) => a.price - b.price);
+}
+//console.log(sortByPrice(data));
+
+
+//4. Сгруппировать продукты по типам. Создать объект, где ключ это тип, а значение - 
+//массив с продуктами.
+
+
+//5. Создать массив с продуктами, которые доступны.
+const newArray = data.filter(function(el) {
+        return el.available === false
+    })
+    //console.log(newArray);
+
+
+
+
+
+
+
+
+//6. Создать функцию, которая принимает массив продуктов и строку = название ингредиента, 
+//и возвращает массив с продуктами, где содержится такой ингредиент.
+const arrWithIngridient = (data, ingridient) => {
```

Comment:
ok!) you can use another solution 
const findProductByIngredient = (products, substring) => products.filter(({ ingredients }) => ingredients.includes(substring));